implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTCallerPrincipal.java

```diff
@@ -164,6 +164,10 @@ private void fixJoseTypes() {
                 replaceNumber(name);
             } else if (claimValue instanceof Boolean) {
                 replaceBoolean(name);
+            } else {
+                // Jose use plain Java types, and we convert stuff to Json. This was added to convert all the missing
+                // types, except String. Without this, we have some inconsistency (apart from String).
+                replaceObject(name);
```

Comment:
@radcortez What other missing types (apart from String) you refer to here ? `Map` covers a complex claim like `Address`, `List` - arrays (of whatever), `Number` - primitive and their wrappers (long/Long, etc), Boolean - booleans.

Reply:
I think I've added this in my previous code where I didn't have the `Boolean` method.

If we look closely, every case can be served by the `replaceObject`, because `JsonUtils.wrapValue` takes an `Object` and returns a `JsonValue`. There is not much point in casting the original claim type or the specific json type. I would propose to drop all `replaceX` methods and keep only `replaceObject`. What do you thin?

Reply:
@radcortez Good point, may be rename `replaceObject` as `replaceClaimValueWithJsonValue` or something like that, I though initially you meant replacing a map inside `replaceObject`. Or may be simply paste the content of `replaceObject` inside the `for loop` iterating over the custom claims, whatever you choose will be fine

(JsonUtils will need some further cleanup but lets deal with it in another issue - specifically arrays and maps will just have Java Integer/Long/etc) so it is only the top level long/etc custom claims are converted to JsonValue. But lets review it again later, to be honest I'd just return simple claim values as is (we already do it for String everywhere, so the same really makes sense for Long/etc) - it is a not a major issue though, we can introduce a property in case some users prefer dealing with JsonValue... But yeah, lets chat about it later

Reply:
Yes. I'll do that change and then we can figure something out with `JsonUtils`. One of the main issues is that our coverage is not great for this cases, so we do need to implement some tests to make sure we don't break.